lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -206,7 +220,49 @@ def auth_approle(self, **kwargs):
         if secret_id is None:
             raise AnsibleError("Authentication method app role requires a secret_id")
 
-        self.client.auth_approle(role_id, secret_id)
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'approle'
+
+        self.client.auth_approle(role_id, secret_id, mount_point=mount_point)
+
+    def auth_gcp(self, **kwargs):
+        role = kwargs.get('role', os.environ.get('VAULT_ROLE', None))
+        if role is None:
+            raise AnsibleError("Authentication method gcp requires a role")
+
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'gcp'
+
+        audience = ANSIBLE_HASHI_VAULT_ADDR + "/vault/" + role
+        headers = {'Metadata-Flavor': 'Google'}
+        format = 'full'
+
+        url = 'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience={0}&format={1}'.format(audience, format)
+        try:
+            jwt = requests.get(url, headers=headers).text
```

Comment:
Use [`fetch_url`](https://github.com/ansible/ansible/blob/240d1a6afb43982f16acebef16778d17aab58160/lib/ansible/module_utils/urls.py#L1197) instead of `requests.get`